scripts/discretization/state_decay_discrete.py

```diff
@@ -29,6 +29,7 @@
 from scipy import optimize
```

Comment:
I'm wondering, why flake8 isn't complaining here. In my editor it says:

    flake8: F401 'scipy.optimize' imported but unused

By the way, there should be a lot of complaints like

    flake8: E303 too many blank lines

Reply:
This is because we haven't added all violation codes. I've added an overview of all possible codes to my initial comment above. One hast to decide now which codes we want to enforce and which ones might be a little bit pedantic :-)

Reply:
I would propose to add various codes step by step and refactor the code accordingly (there is also `autopep8` which might helt with this). Otherwise we might get lost if we refactor everything at once.

Reply:
> Otherwise we might get lost if we refactor everything at once.

That's totally true! So let's do it step by step.